grapher/text/parser.ts

```diff
@@ -0,0 +1,115 @@
+import P, { Language } from "parsimmon"
+
+// An AST inspired by MDAST
+// Deviates because we want to track individual words, whitespace, and newlines to use with TextWrap and our SVG exporter
+
+// The root literal that should be the leaf of every branch that isn't a newline or whitespace
+interface Text {
+    type: "text"
+    value: string
+}
+
+export const textParser = (): P.Parser<Text> =>
+    P.regex(/[A-Za-z'\-":!?,.]+/).map((value) => ({ type: "text", value }))
```

Comment:
Below are all the characters used in subtitles today, and we probably need to support all. Might be better to create a regex that consumes everything apart from some subset of characters?

```
 
!
"
#
$
%
&
'
(
)
+
,
-
.
/
0
1
2
3
4
5
6
7
8
9
:
;
<
=
>
?
A
B
C
D
E
F
G
H
I
J
K
L
M
N
O
P
Q
R
S
T
U
V
W
X
Y
Z
[
\
]
a
b
c
d
e
f
g
h
i
j
k
l
m
n
o
p
q
r
s
t
u
v
w
x
y
z
 
£
­
°
±
²
³
µ
¹
é
ü
 
‑
–
—
‘
’
“
”
•
⁵
₀
₁
₂
₃
₄
₅
−
≥
ﬁ
```

Reply:
This seems like a very annoying problem to solve, since it leads to ambiguity in the parser?

Reply:
Oh boy.

It may be possible to get around this with something else from the toolbox. Some sort of lookahead that chains accordingly. (I'm more on the "just saying words" side of the solution at this point 😅)

Worst case, we can add a more inclusive character regex that still omits characters that will break the parser (`[]*_` etc), and implement error handling that returns the plain string for those errors (because it's safe to assume that any graphers that currently have brackets in them won't have markdown syntax or DoDs currently.)